kornia/utils/misc.py

```diff
@@ -52,3 +56,58 @@ def vec_like(n: int, tensor: Tensor, shared_memory: bool = False) -> Tensor:
 
     vec = zeros(n, 1, device=tensor.device, dtype=tensor.dtype)
     return vec[None].expand(tensor.shape[0], n, 1) if shared_memory else vec[None].repeat(tensor.shape[0], 1, 1)
+
+
+def differentiable_polynomial_rounding(input: Tensor) -> Tensor:
+    """This function implements differentiable rounding.
+
+    Args:
+        input (Tensor): Input tensor of any shape to be rounded.
+
+    Returns:
+        output (Tensor): Pseudo rounded tensor of the same shape as input tensor.
+    """
+    # Perform differentiable rounding
+    output: Tensor = torch.round(input) + (input - torch.round(input)) ** 3
+    return output
+
+
+def differentiable_polynomial_floor(input: Tensor) -> Tensor:
+    """This function implements differentiable floor.
+
+    Args:
+        input (Tensor): Input tensor of any shape to be floored.
+
+    Returns:
+        output (Tensor): Pseudo rounded tensor of the same shape as input tensor.
+    """
+    # Perform differentiable rounding
+    output: Tensor = torch.floor(input) + (input - 0.5 - torch.floor(input)) ** 3
+    return output
+
+
+def differentiable_clipping(
+    input: Tensor,
+    min: Optional[float] = None,
+    max: Optional[float] = None,
+    scale: float = 0.02,
+) -> Tensor:
+    """This function implements a differentiable and soft approximation of the clipping operation.
+
+    Args:
+        input (Tensor): Input tensor of any shape.
+        min (Optional[float]): Minimum value.
+        max (Optional[float]): Maximum value.
+        scale (float): Scale value. Default 0.02.
+
+    Returns:
+        output (Tensor): Clipped output tensor of the same shape as the input tensor.
+    """
+    # Make a copy of the input tensor
+    output: Tensor = input.clone()
+    # Perform differentiable soft clipping
+    if max is not None:
+        output[output > max] = -scale * (torch.exp(-output[output > max] + max) - 1.0) + max
```

Comment:
I think we could try to use clamp/clip with the scaled tensor as arg

Reply:
@edgarriba I didn't get this. Are you talking about `scale` value ? Sorry for asking again

Reply:
something like this (not sure if will work at all) /cc @johnnv1 

```python
if max is not None:
   scaled_value = -scale * ((-output + max_val).exp() - 1.0) + max_val
   output = output.clamp(max=scaled_value)
```
don't use `min` / `max` in the function signature to avoid conflict with python keywords

Reply:
@edgarriba Thank you for explaining. Updated the Changes

Reply:
After the change `tests/enhance/test_jpeg.py::TestDiffJPEG::test_cardinality` and `tests/enhance/test_jpeg.py::TestDiffJPEG::test_gradcheck` are failing due to `Greatest absolute difference: nan at index (0,)` .

Reply:
Hmm, something breaks the differentiability. Since it's the aim of these functions, we should add individual gradcheck tests later. Will check soon what could cause it

Reply:
well, don't know what is the issue with the new code. Torch differentiability breaks easy some times, for me we can keep the original case, and just add tests for gradchecks of each function

Reply:
yes

Reply:
@johnnv1 Updated the code